Xamarin.PropertyEditing.Mac/Controls/FilePathEditorControl.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using AppKit;
+using CoreGraphics;
+using Xamarin.PropertyEditing.Mac.Resources;
+using System.IO;
+using Xamarin.PropertyEditing.Common;
+using Foundation;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class FilePathEditorControl : BasePathEditorControl<FilePath>
+	{
+		public FilePathEditorControl (IHostResourceProvider hostResource)
+		: base (hostResource)
+		{
+			this.currentTextField.ToolTip = this.currentTextField.PlaceholderString = string.Format (LocalizationResources.ChooseFileOrDirectory, LocalizationResources.File);
+			this.panel.CanChooseFiles = true;
+			this.panel.CanChooseDirectories = false;
+			this.revealPathButton.ToolTip = string.Format (LocalizationResources.RevealFileOrDirectory, LocalizationResources.File);
+			this.browsePathButton.ToolTip = string.Format (LocalizationResources.BrowseFileOrDirectory, LocalizationResources.File);
+			this.panel.Title = string.Format (LocalizationResources.ChooseFileOrDirectory, LocalizationResources.File);
+		}
+
+		protected override void OnRevealPathButtonActivated (object sender, EventArgs e)
+		{
+			if (File.Exists (this.currentTextField.StringValue)) {
+				NSWorkspace.SharedWorkspace.OpenFile (this.currentTextField.StringValue);
+			}
+		}
+
+		protected override void StoreCurrentValue ()
+		{
+			if (ViewModel.Value == null) {

```

Comment:
Why would we back out if the existing value is null? This just means we can never set a value to an unset value of this property. If you comment out the initial value setter you have you'll see nothing is set when you try to set it.

Reply:
This was one of the things I fixed in my PR and is what caused the enablement issue.